conan/tools/cmake/presets.py

```diff
@@ -6,26 +6,28 @@
 
 
 def _contents(conanfile, toolchain_file, cache_variables, generator):
-    return {"version": 3,
-            "cmakeMinimumRequired": {"major": 3, "minor": 15, "patch": 0},
-            "configurePresets": [{
+    ret = {"version": 3,
+           "cmakeMinimumRequired": {"major": 3, "minor": 15, "patch": 0},
+           "configurePresets": [{
                 "name": "default",
                 "displayName": "Default Config",
                 "description": "Default configure using '{}' generator".format(generator),
                 "generator": generator,
                 "cacheVariables": cache_variables,
                 "toolchainFile": toolchain_file,
-                "binaryDir": conanfile.build_folder
-            }],
-            "buildPresets": [{
+           }],
+           "buildPresets": [{
               "name": "default",
               "configurePreset": "default"
-            }],
-            "testPresets": [{
+           }],
+           "testPresets": [{
               "name": "default",
               "configurePreset": "default"
-            }]
-            }
+           }]
+          }
+    if conanfile.build_folder:
```

Comment:
A clarifying note saying in which cases the ``build_folder`` is ``None``, but still this presets is being generated would help to understand this condition.